src/components/implementation/tests/chan_evt/chan_evt.c

```diff
@@ -33,22 +33,22 @@ void
 sender(void *d)
 {
 	int i;
-	int idx;
+	word_t idx;
 	
-	idx = (int)d;
+	idx = (word_t)d;
 
 	for (i = 0; i < COMM_AMNT; i++) {
 		thdid_t tid = cos_thdid();
 		assert(tid == test[idx].s_id);
 
 		/* Send to our TID to our own respective channels */
-		if (chan_send(&(test[idx].s), &tid, 0)) {
+		if (chan_send((struct chan_snd *)&(test[idx].s), &tid, 0)) {
```

Comment:
This is also to avoid warnings. I argee with you here, but the compiler just complains on this. This is the warning message:
`warning: passing argument 1 of ‘chan_send’ discards ‘volatile’ qualifier from pointer target type [-Wdiscarded-qualifiers]q

Reply:
maybe the correct design is to remove `volatile` from the array, and place it on the actual struct members where it might be necessary. Here it doesn't matter much as this is a test. If we're doing these casts due to these reasons elsewhere, make a note of the locations, and add an issue with the list of locations to fix.

Reply:
OK

Reply:
https://github.com/gwsystems/composite/blob/3192c8d6cf4bd7134aea8b219ffcb7d7591975a1/src/components/implementation/tests/bench_chan_crt/bench_chan_crt.c#L32
https://github.com/gwsystems/composite/blob/3192c8d6cf4bd7134aea8b219ffcb7d7591975a1/src/components/implementation/tests/chan_evt/chan_evt.c#L27
I went through related commits, these two locations were the changes.
I've removed the `volatile` there.